### PR TITLE
MLFlow project trainer docs

### PR DESCRIPTION
**Pull request by vlad-tokarev**
Docs for new a lightweight "mlflow-project training toolchain"

### CODE REVIEW

File: source/ref_trainings.rst [hunk 1]
```diff
@@ -393,3 +400,53 @@ These names must be `head_input.pkl` and `head_output.pkl`, respectively.
...
+
+MLFlow Project toolchain is a lightweight version of :ref:`MLFlow toolchain`.
+
+The main difference is that MLFlow Project toolchain does not require user stores models using
+`MLFlow Tracking API <https://www.mlflow.org/docs/latest/tracking.html>`_
+and therefore does not require models stored in MLFlow format as a resulted artifact.
+
+Instead, MLFlow Project toolchain relies only on
+`MLFlow Project functionality <https://www.mlflow.org/docs/latest/projects.html>`_  to run training script
+and manage dependencies. User can store result artifacts in any format as he wishes.
```
Comment (keshamin) [inline source/ref_trainings.rst]:
```suggestion
and manage dependencies. User can store result artifacts in any format as they wish.
```

File: source/ref_trainings.rst [hunk 2]
```diff
@@ -393,3 +400,53 @@ These names must be `head_input.pkl` and `head_output.pkl`, respectively.
...
     mlflow.log_artifact('head_output.pkl', 'model')
+
+
+********************************************
+MLFlow Project toolchain
+********************************************
+
+MLFlow Project toolchain is a lightweight version of :ref:`MLFlow toolchain`.
+
+The main difference is that MLFlow Project toolchain does not require user stores models using
```
Comment (keshamin) [inline source/ref_trainings.rst]:
```suggestion
The main difference is that MLFlow Project toolchain does not require user to store models using
```

File: source/ref_trainings.rst [hunk 3]
```diff
@@ -393,3 +400,53 @@ These names must be `head_input.pkl` and `head_output.pkl`, respectively.
...
+
+.. code-block:: python
+    :caption: Example
+    :linenos:
+
+    output_dir = os.environ.get("ODAHUFLOW_OUTPUT_DIR")
+
+    train_x.head().to_pickle(os.path.join(output_dir, 'head_input.pkl'))
+
+Additionally to manual stored artifacts all content from ``$WORK_DIR/data`` folder will be automatically copied to
```
Comment (keshamin) [inline source/ref_trainings.rst]:
```suggestion
Additionally, to manually store artifacts all content from ``$WORK_DIR/data`` folder will be automatically copied to
```

File: source/ref_trainings.rst [hunk 4]
```diff
@@ -393,3 +400,53 @@ These names must be `head_input.pkl` and `head_output.pkl`, respectively.
...
+``$ODAHUFLOW_OUTPUT_DIR``, where ``$WORK_DIR`` is a :ref:`workDir parameter<Training API>` value
+from training spec.
+
+You can use this feature if you have some artifacts that can be defined statically before script execution. For example,
```
Comment (keshamin) [inline source/ref_trainings.rst]:
```suggestion
You can use this feature if you have some file(s) that are required by further steps and can be defined statically before script execution. For example,
```

**Review by keshamin — APPROVED** [review]

File: source/ref_trainings.rst (hunk 1 above)
Comment (karbyshevds) [reply source/ref_trainings.rst]: I'd suggest totally remove `as he wish` and leave it as `User can store result artifacts in any format`. BTW: what does `any format` means? Maybe we should count them?

Comment (vlad-tokarev) [reply source/ref_trainings.rst]: There is no specific set of formats. Just any files

**Review by BPylypenko — APPROVED** [review]

**Pull request merged**